### PR TITLE
Update simplelogin extension

### PR DESCRIPTION
**Pull request by marcelruhf**
## Description
* Convert the Create Random Alias command to a view so that it supports adding a description
* Add 2 new filters, "Show with description" and "Show without description" to allow filtering for aliases with and without descriptions in the List Aliases command
* Cache aliases between command runs for better UX while they are refetched to make the UI interactive immediately

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — COMMENTED** [review]
Looks good to me @marcelruhf 🔥 

Could you check this @marcelruhf

**Comment by marcelruhf** [comment]
> Looks good to me @marcelruhf 🔥
> 
> Could you check this @marcelruhf

You probably meant to ping @adrian-schnell ^ :)

**Comment by pernielsentikaer** [comment]
You're right, thanks for tagging and mentioning 🙂

**Comment by marcelruhf** [comment]
Closing and reopening to retrigger GitHub checks as "Socket Security: Pull Request Alerts" failed.

**Comment by pernielsentikaer** [comment]
@adrian-schnell are you still around, and can you help with the metadata image since it should have the same background 🙂

**Comment by marcelruhf** [comment]
Yay, Socket PR alerts check fixed 🚀

**Comment by marcelruhf** [comment]
> @adrian-schnell are you still around, and can you help with the metadata image since it should have the same background 🙂

@pernielsentikaer If there is no response (or @adrian-schnell indicates they don't have time to maintain it) I'd be happy to recreate all of the screenshots (and take over maintenance of this package if desired)

**Comment by marcelruhf** [comment]
Hey @pernielsentikaer 👋 - just wanted to check in here as I'm wondering what would be the best course of action here? Thanks!

**Comment by pernielsentikaer** [comment]
I tried to ping @adrian-schnell a last time - if no reply I'll test it myself if there is a way to get free simpleplogin account

**Comment by marcelruhf** [comment]
> I tried to ping @adrian-schnell a last time - if no reply I'll test it myself if there is a way to get free simpleplogin account

Sure thing, sounds like a plan!
There is a free tier available, you can sign up here: https://simplelogin.io/

**Comment by pernielsentikaer** [comment]
Awesome, will get it a go in few days so it's not getting all stalled 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

> Convert the Create Random Alias command to a view so that it supports adding a description

This can also be done with `no-view` and an [argument](https://developers.raycast.com/information/lifecycle/arguments#arguments), might be faster to do?

> Cache aliases between command runs for better UX while they are refetched to make the UI interactive immediately

Where is this happening? I tried to create a new alias and went to the list aliases and had to wait for it to fetch it from online 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by marcelruhf** [comment]
> Where is this happening? I tried to create a new alias and went to the list aliases and had to wait for it to fetch it from online 🙂

It does re-fetch them online, but you should notice that after the first fetch when reopening the command it loads noticeably faster (while re-fetching in the background - or at least it does with my 60+ aliases 😄).

**Comment by marcelruhf** [comment]
> Where is this happening? I tried to create a new alias and went to the list aliases and had to wait for it to fetch it from online 🙂

Oh another thng to note - we only cache after the first fetch, not when we create an alias, shoul load from cache when you reopen the list aliases command.

File: extensions/simplelogin/src/list_alias.tsx [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import { useCachedState } from "@raycast/utils";
 import { AliasResponse } from "./models/alias";
 import moment from "moment";
 
-type Filter = "all" | "" | "pinned" | "others";
+type Filter = "all" | "" | "pinned" | "others" | "with-description" | "without-description";
 
 export default function Command() {
-  const [aliases, setAliases] = useState<AliasResponse[]>([]);
+  const [aliases, setAliases] = useCachedState<AliasResponse[]>("aliases", []);
```
Comment (marcelruhf) [inline extensions/simplelogin/src/list_alias.tsx:11]: This is the caching mechanism I introduced here; after the initial fetch, aliases are cached between command restarts.

Comment (pernielsentikaer) [reply extensions/simplelogin/src/list_alias.tsx:11]: Makes sense 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Can you redo the metadata images so we end up with the same background on them all, let me know if you need help

**Comment by marcelruhf** [comment]
@pernielsentikaer I recreated the screenshots as requested
Also sent you a DM on Slack regarding arguments and how they work with hotkeys as there doesn't seem to be a way to supply a value to them when using an assigned hotkey.

File: extensions/simplelogin/src/list_alias.tsx [hunk 2]
```diff
@@ -60,6 +65,7 @@ export default function Command() {
         title: "Are you sure?",
         message: "Do you really want to delete this alias? This action cannot be undone.",
         icon: Icon.DeleteDocument,
+        rememberUserChoice: true,
```
Comment (marcelruhf) [inline extensions/simplelogin/src/list_alias.tsx:68]: I tested this, and if I dismiss the alert and chech the box, I won't be prompted for deletion again.
We probably want this behavior only  when someone decides to go ahead with deletion?

Comment (pernielsentikaer) [reply extensions/simplelogin/src/list_alias.tsx:68]: That's the way it works. If you first tick the "don't show again," it won't pop up again

**Comment by pernielsentikaer** [comment]
Let's just keep it as a view command

**Comment by marcelruhf** [comment]
@pernielsentikaer pushed another commit, ready for review

**Comment by marcelruhf** [comment]
@pernielsentikaer nothing to add here, thanks for the help 😄

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**